multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/event/DynatraceProcessEvent.java

```diff
@@ -0,0 +1,43 @@
+package org.cloudfoundry.multiapps.controller.process.event;
+
+import org.cloudfoundry.multiapps.common.Nullable;
+import org.cloudfoundry.multiapps.controller.api.model.ProcessType;
+import org.cloudfoundry.multiapps.controller.api.model.ProcessTypeDeserializer;
+import org.cloudfoundry.multiapps.controller.api.model.ProcessTypeSerializer;
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableDynatraceProcessEvent.class)
+@JsonDeserialize(as = ImmutableDynatraceProcessEvent.class)
+public abstract class DynatraceProcessEvent {
+    public enum EventType {
+
+        STARTED, FINISHED, FAILED;
+
+        public static EventType fromValue(String v) {
+            for (EventType b : EventType.values()) {
+                if (b.name()
+                     .equals(v)) {
+                    return b;
+                }
+            }
+            return null;
+        }
+    }

```

Comment:
I think it will be more structured if you move this code at the bottom of the file.